pkg/obj/job.go

```diff
@@ -0,0 +1,65 @@
+package obj
+
+import (
+	"context"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Job struct {
+	Key    client.ObjectKey
+	Object *batchv1.Job
+}
+
+var _ Persister = &Job{}
+var _ Loader = &Job{}
+var _ Ownable = &Job{}
+var _ LabelAnnotatableFrom = &Job{}
+
+func (j *Job) Persist(ctx context.Context, cl client.Client) error {
+	if err := Create(ctx, cl, j.Key, j.Object); err != nil {
+		return err
+	}
+
+	_, err := RequiredLoader{j}.Load(ctx, cl)
```

Comment:
Is there a benefit to immediately reloading this object? I thought `Create`/`Update` updated the object with the created version.

Reply:
I will test this again, but I was only seeing updates with the Load present. Otherwise, I was forced to immediately get the object again to see the updates I needed.

Reply:
That may have been a timing issue with specific fields that were legitimately not available yet (i.e. PVC volume name and phase). Either way, I'll verify. With the rush to incorporate the larger functionality, I haven't gone back and looked over some of these details again. Fairly glad I decided to split these code changes up.